frontend/js/src/common/brainzplayer/AppleMusicPlayer.tsx

```diff
@@ -148,7 +149,11 @@ export default class AppleMusicPlayer
   }
 
   componentDidUpdate(prevProps: DataSourceProps) {
-    const { show } = this.props;
+    const { show, volume } = this.props;
+    const player = this.appleMusicPlayer;
+    if (player) {
```

Comment:
For each of these calls in componentDidUpdate, you need to check if the `volume` prop has changed:
```suggestion
    if (prevProps.volume !== volume && player) {
```